ledger/acctupdates.go

```diff
@@ -350,181 +375,217 @@ func (au *accountUpdates) ListApplications(maxAppIdx basics.AppIndex, maxResults
 // listCreatables lists the application/asset by their app/asset index, limiting to the first maxResults
 func (au *accountUpdates) listCreatables(maxCreatableIdx basics.CreatableIndex, maxResults uint64, ctype basics.CreatableType) ([]basics.CreatableLocator, error) {
 	au.accountsMu.RLock()
-	defer au.accountsMu.RUnlock()
-
-	// Sort indices for creatables that have been created/deleted. If this
-	// turns out to be too inefficient, we could keep around a heap of
-	// created/deleted asset indices in memory.
-	keys := make([]basics.CreatableIndex, 0, len(au.creatables))
-	for cidx, delta := range au.creatables {
-		if delta.ctype != ctype {
-			continue
+	for {
+		currentDbRound := au.dbRound
+		currentDeltaLen := len(au.deltas)
+		// Sort indices for creatables that have been created/deleted. If this
+		// turns out to be too inefficient, we could keep around a heap of
+		// created/deleted asset indices in memory.
+		keys := make([]basics.CreatableIndex, 0, len(au.creatables))
+		for cidx, delta := range au.creatables {
+			if delta.ctype != ctype {
+				continue
+			}
+			if cidx <= maxCreatableIdx {
+				keys = append(keys, cidx)
+			}
 		}
-		if cidx <= maxCreatableIdx {
-			keys = append(keys, cidx)
+		sort.Slice(keys, func(i, j int) bool { return keys[i] > keys[j] })
+
+		// Check for creatables that haven't been synced to disk yet.
+		var unsyncedCreatables []basics.CreatableLocator
+		deletedCreatables := make(map[basics.CreatableIndex]bool)
+		for _, cidx := range keys {
+			delta := au.creatables[cidx]
+			if delta.created {
+				// Created but only exists in memory
+				unsyncedCreatables = append(unsyncedCreatables, basics.CreatableLocator{
+					Type:    delta.ctype,
+					Index:   cidx,
+					Creator: delta.creator,
+				})
+			} else {
+				// Mark deleted creatables for exclusion from the results set
+				deletedCreatables[cidx] = true
+			}
 		}
-	}
-	sort.Slice(keys, func(i, j int) bool { return keys[i] > keys[j] })
 
-	// Check for creatables that haven't been synced to disk yet.
-	var unsyncedCreatables []basics.CreatableLocator
-	deletedCreatables := make(map[basics.CreatableIndex]bool)
-	for _, cidx := range keys {
-		delta := au.creatables[cidx]
-		if delta.created {
-			// Created but only exists in memory
-			unsyncedCreatables = append(unsyncedCreatables, basics.CreatableLocator{
-				Type:    delta.ctype,
-				Index:   cidx,
-				Creator: delta.creator,
-			})
-		} else {
-			// Mark deleted creatables for exclusion from the results set
-			deletedCreatables[cidx] = true
+		// Check in-memory created creatables, which will always be newer than anything
+		// in the database
+		var res []basics.CreatableLocator
```

Comment:
nit: `res = append(res, loc[:max(maxResults, len(loc)]...)`  + cond. I mean we can get rid loop here.

Reply:
that's a good point, and I'll do that next in a separate PR.